astropy/io/ascii/ui.py

```diff
@@ -1059,10 +1058,14 @@ def write(
         # behavior is for Python to translate \r\n (which we write because
         # of os.linesep) into \r\r\n. Specifying newline='' disables any
         # auto-translation.
-        output = open(output, "w", newline="")
-        output.write(outstr)
-        output.write(os.linesep)
-        output.close()
+        if not output:
```

Comment:
But there is already this check above:

```python
    if output is None:
        output = sys.stdout
```

Reply:
This check is different: it also excludes the empty string `''` because `Path('')` means "the current work directory", and this scenario (passing `output=''` and expecting a `FileNotFoundError`) is actually exercised in a test.

Reply:
What was the old behavior for `output=''`? Does this need a change log?

Reply:
Au contraire, this is being strictly backward compatible: `open('')` raises the same exception.

Reply:
I am seeing this error message:
```
>>> open("", "w")
FileNotFoundError: [Errno 2] No such file or directory: ''
```

Reply:
I'm going to revert this and simply refactor with a context manager as suggested in https://github.com/astropy/astropy/pull/16954/files#r1748045678